akka-actor-typed-tests/src/test/scala/akka/actor/typed/AskSpec.scala

```diff
@@ -111,6 +119,63 @@ class AskSpec extends ScalaTestWithActorTestKit("""
       }
     }
 
+    "fail the future if the actor has terminated" in {
+      implicit val timeout = Timeout(10.millis)
+      val actor: ActorRef[Msg] = spawn(behavior)
+
+      val deadLetterProbe = createDeadLetterProbe()
+      val probe = createTestProbe[Any]()
+      actor ! Stop(probe.ref)
+
+      val answer: Future[String] = actor.ask(Foo("bar", _))
+      val result = answer.failed.futureValue
+      result shouldBe a[TimeoutException]
+      result.getMessage should startWith("Ask timed out on")
```

Comment:
This test seems flaky... there's a race between the actor between stopping and the ask.  If one side wins, the message will start with this, but if the other wins, it will start with "had already been terminated."

Reply:
https://github.com/akka/akka/pull/32111 addresses

Reply:
Yep, It is also has a fix commit on #32107